### PR TITLE
fix: Properly scale images and respect cache behavior

### PR DESCRIPTION
**Pull request by stormofice**
Fixes two issues:
- When an image was found in the cache, it did not matter whether CacheBehavior was set to Skip or not, it would get used anyway
- When scaling images, the height was always provided as a ThumbnailOption even if it was not required. This lead to VIPS keeping that height and not performing any scaling.

NOTE: The same issue is still present for the height parameter, e.g. try to fetch any image with ?height=XYZ. It has no scaling applied to it. However, I do not see a way to fix this with the libvips bindings.

Previously, images were not scaled but cut, i.e. some parts were cut off instead of the whole image being present in the result.

### CODE REVIEW

**Comment by GamerBene19** [comment]
Closing in favor of #28.

TL;DR: Removing crop option (which is equal to setting it to None) fixes the issue and also handles height correctly.

I kept the change regarding cache behavior in #28 and added you as co author.
